test/functional/rpc_whitelist.py

```diff
@@ -92,5 +103,53 @@ def run_test(self):
         self.log.info("Strange test 5")
         assert_equal(200, rpccall(self.nodes[0], self.strange_users[4], "getblockcount").status)
 
+        self.test_rpcwhitelistdefault_0_no_permissions()
+        self.test_rpcwhitelistdefault_1_no_permissions()
+        self.test_rpcwhitelistdefault_1_with_user_permissions()
+
+    def test_rpcwhitelistdefault_0_no_permissions(self):
+        """
+        * rpcwhitelistdefault=0
+        * No Permissions defined
+        Expected result: * user3 (not whitelisted) can access any method
+        """
+        unrestricted_user = self.users[2]
+        for permission in self.COMMON_PERMISSIONS:
+            self.log.info("[" + unrestricted_user[0] + "]: Testing unrestricted user permission (" + permission + ")")
+            assert_equal(200, rpccall(self.nodes[0], unrestricted_user, permission).status)
+
+    def test_rpcwhitelistdefault_1_no_permissions(self):
+        """
+        * rpcwhitelistdefault=1
+        * No Permissions defined
+        Expected result: * user3 (not whitelisted) can not access any method
+        """
+        # Replace file configurations
+        self.nodes[0].replace_in_config([("rpcwhitelistdefault=0", "rpcwhitelistdefault=1")])
+        with open(self.nodes[0].datadir_path / "bitcoin.conf", 'a', encoding='utf8') as f:
+            f.write("rpcwhitelist=__cookie__:getblockcount,getblockchaininfo,getmempoolinfo,stop\n")
+        self.restart_node(0)
```

Comment:
In commit "test: Add test for rpcwhitelistdefault" (fcf0ead6cd358fd35909e1102bdae2c176b0ace6)

Since this code is changing configuration that affects all future tests I think it would be less confusing if it were moved out of this individual test method and into the main `run_test` method. Otherwise it is not clear that calling this particular test method has side effects for calls to future test methods.